client/lib/user/user.d.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+import { Language } from '../../languages';
+import { GravatarOptions } from '../../gravatar/types';
+import { URL, JSONSerializable } from '../../types';
+
+type WPCOMError = { message: string };
+type WPCOMHeaders = { [ name: string ]: string } & { status: number };
+
+type WPCOMCallback = (
+	error: WPCOMError | null,
+	result: JSONSerializable | null,
+	headers: WPCOMHeaders
+) => void;
+
+export type UserMetaData = {
+	links: Record< 'self' | 'help' | 'site' | 'flags', URL >;
+	data?: {
+		flags: {
+			active_flags: string[];
+		};
+	};
+};
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: ( userId: number ) => void;
+	get: () => UserData;
+	fetch: () => Promise< void >;
```

Comment:
just pointing out how the types help with comprehension here. seeing `Promise<void>` makes it clear that I shouldn't expect a `User` from this call. pretty cool!